<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>sudoku_game</class>
 <widget class="QMainWindow" name="sudoku_game">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>862</width>
    <height>637</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SUDOKU</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>581</width>
      <height>581</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <widget class="QTableWidget" name="sudoku_table">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>561</width>
       <height>561</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::AnyKeyPressed|QAbstractItemView::DoubleClicked|QAbstractItemView::EditKeyPressed|QAbstractItemView::SelectedClicked</set>
     </property>
     <property name="iconSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="rowCount">
      <number>9</number>
     </property>
     <property name="columnCount">
      <number>9</number>
     </property>
     <attribute name="horizontalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>60</number>
     </attribute>
     <attribute name="horizontalHeaderHighlightSections">
      <bool>true</bool>
     </attribute>
     <attribute name="horizontalHeaderMinimumSectionSize">
      <number>20</number>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="verticalHeaderDefaultSectionSize">
      <number>60</number>
     </attribute>
     <attribute name="verticalHeaderMinimumSectionSize">
      <number>25</number>
     </attribute>
     <attribute name="verticalHeaderShowSortIndicator" stdset="0">
      <bool>true</bool>
     </attribute>
     <attribute name="verticalHeaderStretchLastSection">
      <bool>false</bool>
     </attribute>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
    </widget>
    <widget class="Line" name="line">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>192</y>
       <width>538</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="Line" name="line_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>372</y>
       <width>538</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="Line" name="line_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>15</y>
       <width>545</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="Line" name="line_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>553</y>
       <width>545</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="Line" name="line_5">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>23</y>
       <width>20</width>
       <height>534</height>
      </rect>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="Line" name="line_6">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>23</y>
       <width>20</width>
       <height>534</height>
      </rect>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="Line" name="line_7">
     <property name="geometry">
      <rect>
       <x>12</x>
       <y>23</y>
       <width>20</width>
       <height>536</height>
      </rect>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="Line" name="line_8">
     <property name="geometry">
      <rect>
       <x>552</x>
       <y>23</y>
       <width>20</width>
       <height>536</height>
      </rect>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_7">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>280</y>
      <width>251</width>
      <height>311</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>231</width>
       <height>161</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="1">
       <widget class="QPushButton" name="button_2">
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="button_3">
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>3</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="button_4">
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>4</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="button_6">
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>6</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="button_5">
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>5</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="button_1">
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="button_7">
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>7</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="button_8">
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>8</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="button_9">
        <property name="maximumSize">
         <size>
          <width>72</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>9</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>210</y>
       <width>231</width>
       <height>91</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="1" column="0">
       <widget class="QPushButton" name="reset_game">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>31</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>RESET GAME</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="undo">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>31</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>UNDO PREVIOUS STEP</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>56</x>
       <y>10</y>
       <width>141</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>NUMBER ENTRY</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>150</y>
      <width>251</width>
      <height>121</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QPushButton" name="check_errors">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>231</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>CHECK ERRORS</string>
     </property>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>231</width>
       <height>61</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>10</pointsize>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>GAME TIME:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="timer_display">
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>00:00:00</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="font">
         <font>
          <family>Nirmala UI</family>
          <pointsize>10</pointsize>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>ERRORS:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="errors_display">
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_3">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>10</y>
      <width>251</width>
      <height>131</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>CUSTOM GAME</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QWidget" name="formLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>225</width>
       <height>70</height>
      </rect>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string># OF HIDDEN SQUARES:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="custom_hidden">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>31</height>
         </size>
        </property>
        <property name="maximum">
         <number>81</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QPushButton" name="start_custom_game">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>31</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>START NEW GAME</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>862</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuNEW_GAME">
    <property name="title">
     <string>NEW GAME</string>
    </property>
    <addaction name="start_easy_game"/>
    <addaction name="start_medium_game"/>
    <addaction name="start_hard_game"/>
    <addaction name="start_stupes_game"/>
   </widget>
   <addaction name="menuNEW_GAME"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="start_easy_game">
   <property name="text">
    <string>Easy</string>
   </property>
  </action>
  <action name="start_medium_game">
   <property name="text">
    <string>Medium</string>
   </property>
  </action>
  <action name="start_hard_game">
   <property name="text">
    <string>Hard</string>
   </property>
  </action>
  <action name="start_stupes_game">
   <property name="text">
    <string>Stupes</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
